src/NUnitEngine/nunit.engine/Internal/ResultHelper.cs

```diff
@@ -138,15 +145,15 @@ public static XmlNode Aggregate(string elementName, string testType, string name
 
             string status = "Inconclusive";
             //double totalDuration = 0.0d;
-            int testcasecount = 0;
-            int total = 0;
-            int passed = 0;
-            int failed = 0;
-            int inconclusive = 0;
-            int skipped = 0;
-            int asserts = 0;
+            var testcasecount = 0;
+            var total = 0;
+            var passed = 0;
+            var failed = 0;
+            var inconclusive = 0;
+            var skipped = 0;
+            var asserts = 0;
```

Comment:
I agree,  var should never be used for primitive types, and should only be used when the right side of the = specifically specifies the type in some fashion (cast, as, new, …)

 

From: CharliePoole [mailto:notifications@github.com] 
Sent: Thursday, April 16, 2015 1:55 AM
To: nunit/nunit
Subject: Re: [nunit] Suggested fix for issue 22 (#591)

 

In src/NUnitEngine/nunit.engine/Internal/ResultHelper.cs <https://github.com/nunit/nunit/pull/591#discussion_r28444811> :

> @@ -138,15 +145,15 @@ public static XmlNode Aggregate(string elementName, string testType, string name
>  
>              string status = "Inconclusive";
>              //double totalDuration = 0.0d;
> -            int testcasecount = 0;
> -            int total = 0;
> -            int passed = 0;
> -            int failed = 0;
> -            int inconclusive = 0;
> -            int skipped = 0;
> -            int asserts = 0;
> +            var testcasecount = 0;
> +            var total = 0;
> +            var passed = 0;
> +            var failed = 0;
> +            var inconclusive = 0;
> +            var skipped = 0;
> +            var asserts = 0;

Remember we had comments from several people about not using var for built-in types. I'm in agreement and I'll incorporate that in an edit of the standards shortly.

—
Reply to this email directly or view it on GitHub <https://github.com/nunit/nunit/pull/591/files#r28444811> .  <https://github.com/notifications/beacon/AFh64UC5koZTSx-kvCjWFYHgcCT9lwwOks5n_p11gaJpZM4EAxBH.gif>